From 9ff2df14e09436b844738d403ef01629f7e3b608 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <xhochy@users.noreply.github.com>
Date: Tue, 21 Jan 2020 12:33:36 +0100
Subject: [PATCH] Fixes for conda build with Arrow support

---
 setup.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 0da6a8a..51cb4f5 100644
--- a/setup.py
+++ b/setup.py
@@ -85,8 +85,6 @@ if isBuildExtEnabled == 'true':
             current_dir = os.getcwd()
 
             if ext.name == 'snowflake.connector.arrow_iterator':
-                self._copy_arrow_lib()
-
                 ext.sources += ['cpp/ArrowIterator/CArrowIterator.cpp',
                                 'cpp/ArrowIterator/CArrowChunkIterator.cpp',
                                 'cpp/ArrowIterator/CArrowTableIterator.cpp',
@@ -113,8 +111,6 @@ if isBuildExtEnabled == 'true':
                 elif platform == 'linux' or platform == 'darwin':
                     ext.extra_compile_args.append('-isystem' + pyarrow.get_include())
                     ext.extra_compile_args.append('-isystem' + numpy.get_include())
-                    ext.extra_compile_args.append('-std=c++11')
-                    ext.extra_compile_args.append('-D_GLIBCXX_USE_CXX11_ABI=0')
 
                 ext.library_dirs.append(os.path.join(current_dir, self.build_lib, 'snowflake', 'connector'))
                 ext.extra_link_args += self._get_arrow_lib_as_linker_input()
@@ -132,7 +128,7 @@ if isBuildExtEnabled == 'true':
             build_ext.build_extension(self, ext)
 
         def _get_arrow_lib_dir(self):
-            return pyarrow.get_library_dirs()[0]
+            return os.path.join(os.environ['PREFIX'], 'lib')
 
         def _copy_arrow_lib(self):
             libs_to_bundle = self.arrow_libs_to_copy[sys.platform]
-- 
2.24.0

