From 14393a6f8db055674ca247497c16342bf7edd1b2 Mon Sep 17 00:00:00 2001
From: Felix Hoehle <fhoehle@users.noreply.github.com>
Date: Sun, 19 Dec 2021 16:01:43 +0100
Subject: [PATCH] drop pyarrow version numbers

---
 pyproject.toml |  2 +-
 setup.py       | 10 +++++-----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index edf8aac..6e62310 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,5 +6,5 @@ requires = [
     "wheel",
     "cython",
     # Must be kept in sync with the `setup_requirements` in `setup.py`
-    "pyarrow>=6.0.0,<6.1.0",
+    "pyarrow",
 ]
diff --git a/setup.py b/setup.py
index ab1b2db..47dbc37 100644
--- a/setup.py
+++ b/setup.py
@@ -48,7 +48,7 @@ cmd_class = {}
 
 pandas_requirements = [
     # Must be kept in sync with pyproject.toml
-    "pyarrow>=6.0.0,<6.1.0",
+    "pyarrow",
     "pandas>=1.0.0,<1.4.0",
 ]
 
@@ -80,14 +80,14 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
         # this list should be carefully examined when pyarrow lib is
         # upgraded
         arrow_libs_to_copy = {
-            "linux": ["libarrow.so.600", "libarrow_python.so.600"],
-            "darwin": ["libarrow.600.dylib", "libarrow_python.600.dylib"],
+            "linux": ["libarrow.so", "libarrow_python.so"],
+            "darwin": ["libarrow.dylib", "libarrow_python.dylib"],
             "win32": ["arrow.dll", "arrow_python.dll"],
         }
 
         arrow_libs_to_link = {
-            "linux": ["libarrow.so.600", "libarrow_python.so.600"],
-            "darwin": ["libarrow.600.dylib", "libarrow_python.600.dylib"],
+            "linux": ["libarrow.so", "libarrow_python.so"],
+            "darwin": ["libarrow.dylib", "libarrow_python.dylib"],
             "win32": ["arrow.lib", "arrow_python.lib"],
         }
 
-- 
2.34.1

