From 36485789264c01a6706ef05ed6237f0d0726a23c Mon Sep 17 00:00:00 2001
From: Adam Ling <adam.ling@snowflake.com>
Date: Mon, 29 Jan 2024 16:56:43 -0800
Subject: [PATCH] bump dependency

---
 DESCRIPTION.md | 5 +++++
 setup.cfg      | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/DESCRIPTION.md b/DESCRIPTION.md
index 8de5d5e3d..fe98408ea 100644
--- a/DESCRIPTION.md
+++ b/DESCRIPTION.md
@@ -8,6 +8,11 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
+- v3.7.1
+
+  - Bumped cryptography dependency from <42.0.0,>=3.1.0 to >=3.1.0,<43.0.0.
+  - Bumped pyOpenSSL dependency from >=16.2.0,<24.0.0 to >=16.2.0,<25.0.0.
+
 - v3.7.0(January 25,2024)
 
   - Added a new boolean parameter `force_return_table` to `SnowflakeCursor.fetch_arrow_all` to force returning `pyarrow.Table` in case of zero rows.
diff --git a/setup.cfg b/setup.cfg
index 4012abe04..e335fb1a5 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -45,8 +45,8 @@ packages = find_namespace:
 install_requires =
     asn1crypto>0.24.0,<2.0.0
     cffi>=1.9,<2.0.0
-    cryptography>=3.1.0,<42.0.0
-    pyOpenSSL>=16.2.0,<24.0.0
+    cryptography>=3.1.0,<43.0.0
+    pyOpenSSL>=16.2.0,<25.0.0
     pyjwt<3.0.0
     pytz
     requests<3.0.0
