From 8ee7b49a443a8e495946a9f8714d3d1e16c24b8b Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 18 Sep 2020 19:51:12 +0200
Subject: [PATCH] Fixes for conda build with Arrow support

---
 setup.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 6d6b8d3..6094eee 100644
--- a/setup.py
+++ b/setup.py
@@ -96,7 +96,6 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
             current_dir = os.getcwd()
 
             if ext.name == 'snowflake.connector.arrow_iterator':
-                self._copy_arrow_lib()
                 CPP_SRC_DIR = os.path.join(CONNECTOR_SRC_DIR, 'cpp')
                 ARROW_ITERATOR_SRC_DIR = os.path.join(CPP_SRC_DIR, 'ArrowIterator')
                 LOGGING_SRC_DIR = os.path.join(CPP_SRC_DIR, 'Logging')
@@ -127,8 +126,6 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
                 elif platform == 'linux' or platform == 'darwin':
                     ext.extra_compile_args.append('-isystem' + pyarrow.get_include())
                     ext.extra_compile_args.append('-isystem' + numpy.get_include())
-                    ext.extra_compile_args.append('-std=c++11')
-                    ext.extra_compile_args.append('-D_GLIBCXX_USE_CXX11_ABI=0')
 
                 ext.library_dirs.append(os.path.join(current_dir, self.build_lib, 'snowflake', 'connector'))
                 ext.extra_link_args += self._get_arrow_lib_as_linker_input()
@@ -146,7 +143,7 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
             build_ext.build_extension(self, ext)
 
         def _get_arrow_lib_dir(self):
-            return pyarrow.get_library_dirs()[0]
+            return os.path.join(os.environ['PREFIX'], 'lib')
 
         def _copy_arrow_lib(self):
             libs_to_bundle = self.arrow_libs_to_copy[sys.platform]
-- 
2.28.0

