From ba0ede7638e1d78c72339784f7918202c907f249 Mon Sep 17 00:00:00 2001
From: fhoehle <fhoehle@users.noreply.github.com>
Date: Fri, 27 Aug 2021 16:02:29 +0200
Subject: [PATCH] drop pyarrow version numbers

---
 setup.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 562b6bd..7fd564a 100644
--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,7 @@ cmd_class = {}
 
 pandas_requirements = [
     # Must be kept in sync with pyproject.toml
-    "pyarrow>=5.0.0,<5.1.0",
+    "pyarrow",
     "pandas>=1.0.0,<1.3.0",
 ]
 
@@ -83,14 +83,14 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
         # this list should be carefully examined when pyarrow lib is
         # upgraded
         arrow_libs_to_copy = {
-            "linux": ["libarrow.so.500", "libarrow_python.so.500"],
-            "darwin": ["libarrow.500.dylib", "libarrow_python.500.dylib"],
+            "linux": ["libarrow.so", "libarrow_python.so"],
+            "darwin": ["libarrow.dylib", "libarrow_python.dylib"],
             "win32": ["arrow.dll", "arrow_python.dll"],
         }
 
         arrow_libs_to_link = {
-            "linux": ["libarrow.so.500", "libarrow_python.so.500"],
-            "darwin": ["libarrow.500.dylib", "libarrow_python.500.dylib"],
+            "linux": ["libarrow.so", "libarrow_python.so"],
+            "darwin": ["libarrow.dylib", "libarrow_python.dylib"],
             "win32": ["arrow.lib", "arrow_python.lib"],
         }
 
-- 
2.30.2

