From d028d5223b5731e0b2b5f646b461eae11d94b077 Mon Sep 17 00:00:00 2001
From: Andrew Achkar <hajapy@users.noreply.github.com>
Date: Fri, 28 Oct 2022 23:02:06 -0400
Subject: [PATCH] Bump cryptography version to <39.0.0

---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index b714f40..65edfbd 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -44,7 +44,7 @@ packages = find_namespace:
 install_requires =
     asn1crypto>0.24.0,<2.0.0
     cffi>=1.9,<2.0.0
-    cryptography>=3.1.0,<37.0.0
+    cryptography>=3.1.0,<39.0.0
     oscrypto<2.0.0
     pyOpenSSL>=16.2.0,<23.0.0
     pycryptodomex!=3.5.0,>=3.2,<4.0.0
--
2.37.0 (Apple Git-136)
