From 837f9a12a72985789d58a235b321c47628e324fe Mon Sep 17 00:00:00 2001
From: Marius van Niekerk <marius.v.niekerk@gmail.com>
Date: Tue, 9 Feb 2021 16:15:04 -0500
Subject: [PATCH 3/3] Improve setup.py for multiple arrow versions

---
 pyproject.toml |  2 +-
 setup.py       | 31 ++++++++++++++++++-------------
 2 files changed, 19 insertions(+), 14 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index a58827c..5f655da 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,5 +6,5 @@ requires = [
     "wheel",
     "cython",
     # Must be kept in sync with the `setup_requirements` in `setup.py`
-    'pyarrow>=2.0.0,<2.1.0',
+    'pyarrow>=2.0.0,<4.0',
 ]
diff --git a/setup.py b/setup.py
index 5ef61cf..9824bca 100644
--- a/setup.py
+++ b/setup.py
@@ -41,7 +41,7 @@ cmd_class = {}
 
 pandas_requirements = [
     # Must be kept in sync with pyproject.toml
-    'pyarrow>=2.0.0,<2.1.0',
+    'pyarrow>=2.0.0,<4.0.0',
     'pandas>=1.0.0,<1.2.0',
 ]
 
@@ -71,22 +71,27 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
         # list of libraries that will be bundled with python connector,
         # this list should be carefully examined when pyarrow lib is
         # upgraded
+        try:
+            so_ver = pyarrow.cpp_build_info.so_version
+        except:
+            # fallback to pyarrow 0.17
+            so_ver = 17
         arrow_libs_to_copy = {
-            'linux': ['libarrow.so.200',
-                      'libarrow_python.so.200',
-                      'libarrow_flight.so.200'],
-            'darwin': ['libarrow.200.dylib',
-                       'libarrow_python.200.dylib'],
+            'linux': [f'libarrow.so.{so_ver}',
+                      f'libarrow_python.so.{so_ver}',
+                      f'libarrow_flight.so.{so_ver}'],
+            'darwin': [f'libarrow.{so_ver}.dylib',
+                       f'libarrow_python.{so_ver}.dylib'],
             'win32': ['arrow.dll',
                       'arrow_python.dll',
                       'zlib.dll']
         }
 
         arrow_libs_to_link = {
-            'linux': ['libarrow.so.200',
-                      'libarrow_python.so.200'],
-            'darwin': ['libarrow.200.dylib',
-                       'libarrow_python.200.dylib'],
+            'linux': [f'libarrow.so.{so_ver}',
+                      f'libarrow_python.so.{so_ver}'],
+            'darwin': [f'libarrow.{so_ver}.dylib',
+                       f'libarrow_python.{so_ver}.dylib'],
             'win32': ['arrow.lib',
                       'arrow_python.lib']
         }
@@ -155,7 +160,7 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
             libs_to_bundle = self.arrow_libs_to_copy[sys.platform]
 
             for lib in libs_to_bundle:
-                source = '{}/{}'.format(self._get_arrow_lib_dir(), lib)
+                source = os.path.join(self._get_arrow_lib_dir(), lib)
                 build_dir = os.path.join(self.build_lib, 'snowflake', 'connector')
                 copy(source, build_dir)
 
@@ -164,8 +169,8 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
             ret = []
 
             for lib in link_lib:
-                source = '{}/{}'.format(self._get_arrow_lib_dir(), lib)
-                assert os.path.exists(source)
+                source = os.path.join(self._get_arrow_lib_dir(), lib)
+                assert os.path.exists(source), f"`{source}` does not exist!"
                 ret.append(source)
 
             return ret
-- 
2.30.0

